src/modules/esl-media-query/README.md

```diff
@@ -126,22 +79,96 @@ class Test {
+
+You can get all available screen breakpoints using the `ESLScreenBreakpoints.names` property or access the complete breakpoint definitions with the `ESLScreenBreakpoints.get` method, which will return an `ESLScreenBreakpoint` instance if it exists.
+
+The list of predefined breakpoint is the following:
+  - `@xs` from 1 to 767 pixels
+  - `@sm` from 768 to 991 pixels
+  - `@md` from 992 to 1199 pixels
+  - `@lg` from 1200 to 1599 pixels
+  - `@xl` from 1600 to 999999 pixels
+
+---
+<a name="dpr"></a>
+### ESLScreenDPR
+
+Additional preprocessor to provide DPR shortcuts.
+
+#### ESLScreenDPR API
+  - `ignoreBotsDpr` - marker to enable DPR ignoring for PageSpeed Bots
+  - `toDPI` - converts to DPI (Dots Per Inch) based on a given number value
+  - `process` - creates a string representation of the DPI value, in the format of (e.g. `(min-resolution: {value}dpi)`)
+
+---
+
+<a name="shortcuts"></a>
+### ESLEnvShortcuts
+
+ESLEnvShortcuts is a simple registry for a static shortcuts to describe environment related conditions
+
+An additional shortcuts can be added to the registry through `ESLEnvShortcuts.add` method
+ESLEnvShortcuts allows adding boolean result that will be converted to `all` / `not all` query conditions or setup result as native MediaQuery string. 
+
+Environment shortcuts can be removed with `ESLEnvShortcuts.remove` method.
+
+The list of predefined environment shortcuts is the following:
+  - `@BOT` - detects search/pagespeed bots
+  - `@MOBILE` - detects mobile devices 
+  - `@DESKTOP` - detects desktops
+  - `@ANDROID` - detects Android devices
+  - `@IOS` - detects iOS devices  
+  - `@TOUCH` - cross-browser touch support detection
+  - `@IE` - detects InternetExplorer 11
+  - `@EDGE` - detects legacy Edge browser (<18) based on EdgeHTML engine
+  - `@GECKO` - detects Gecko engine based browsers (e.g. Firefox)
+  - `@BLINK` - detects Blink engine based browsers (e.g. Google Chrome, Opera)
+  - `@SAFARI` - detects safari browsers
+  - `@SAFARI-IOS` - detects mobile safari browsers
+
 ---
 
 <a name="media-rule"></a>
 ### ESLMediaRule
 
 Pair of `ESLMediaQuery` and value (payload) associated with a query condition. 
-`ESLMediaRule` is used as an item for [ESLMediaRuleList](#eslmediarulelist).  
+`ESLMediaRule` is used as an item for [ESLMediaRuleList](#media-rule-list).  
 `ESLMediaRule` can be parsed from `<ESL Media Query> => <value>` syntax string, 
 e.g. `@XS => 1` (`1` is the payload) or `@+LG and @DESKTOP => desktop` (`'desktop'` is the payload).
 
+#### ESLMediaRule API
+- `ESLMediaRule.prototype.create(payload, query, parser)` - creates an `ESLMediaRule` instance from a payload string, media query, and a payload parser function. If the payload parsing result is undefined, the rule will also be undefined
+- `ESLMediaRule.prototype.parse` - parses media rule defined with syntax, where query and value separated by `=>`
+- `ESLMediaRule.prototype.all(payload)` - creates an always active `ESLMediaRule` with passed value
```

Comment:
mb will be better 'always' place at the start of phrase

Reply:
sounds okay both ways for me